File: src/main/java/wraith/fabricaeexnihilo/api/recipes/barrel/FluidTransformRecipe.java
package wraith.fabricaeexnihilo.api.recipes.barrel;

import wraith.fabricaeexnihilo.api.crafting.FluidIngredient;
import wraith.fabricaeexnihilo.api.crafting.ItemIngredient;
import wraith.fabricaeexnihilo.modules.barrels.modes.BarrelMode;

public record FluidTransformRecipe(FluidIngredient inBarrel, ItemIngredient catalyst, BarrelMode result) {
}

File: src/main/kotlin/exnihilofabrico/api/recipes/barrel/FluidTransformRecipe.kt
package exnihilofabrico.api.recipes.barrel

import exnihilofabrico.api.crafting.FluidIngredient
import exnihilofabrico.api.crafting.ItemIngredient
import exnihilofabrico.modules.barrels.modes.BarrelMode

data class FluidTransformRecipe(val inBarrel: FluidIngredient, val catalyst: ItemIngredient, val result: BarrelMode)